### PR TITLE
fix(scaffolder): Allow user to retry creating new component without having to reload the page

### PR DESCRIPTION
**Pull request by OrkoHunter**
# Problem

output

(30 seconds GIF)

If the job of creating a new component fails due to some reason and if the user wants to retry, the "Submit" button does not work. Even after doing a "Reset", the "Submit" button is unresponsive.

# How to Reproduce

A simple way to encounter a failed job scenario might be closing docker application and trying to create a new component.

# Technical issue (and fix)

Underneath the UI, the "Submit" button _does_ actually create a new job. There are two problems - New modal doesn't show up and the Polling mechanism doesn't pick up the new job ID.

1. When the user clicks the `Close` button on the `JobStatusModal`, it does not set the `modalOpen` state of its parent component. Thus, even after closing, `JobStatusModal` still persists in the DOM, and `modalOpen` is still set to `true`. That is why, retrying by clicking on "Submit" again does not reopen the modal.

2. The `useJobPolling` hook does not consider restarting the timer again when the `job` changes. Even if `jobId` is set to the new value, the logic to set `shouldBeRunningInterval` still uses a previous value of `currentJob` since it is not being reset after the timer stops. Using an effect hook and having `jobId` as a dependency to reset `currentJob` is the way to fix this.

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by O5ten** [comment]
Verified working on our rig! Thanks @OrkoHunter !

File: plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx [hunk 1]
```diff
@@ -23,18 +23,23 @@ import {
...
-import React, { useCallback, useState } from 'react';
+import React, { useCallback } from 'react';
 import { Job } from '../../types';
 import { JobStage } from '../JobStage/JobStage';
 
 type Props = {
   job: Job | null;
   toCatalogLink?: string;
+  open: boolean;
+  setOpen: (newState: boolean) => void;
```
Comment (Rugvip) [inline plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx:34]: Feeling it makes more sense to make this `onClose`, you usually don't pass down setters as props

**Review by Rugvip — COMMENTED** [review]
Thanks! A nit but pretty much good to go :>

**Comment by OrkoHunter** [comment]
@Rugvip @freben cc @benjdlambert Thanks for the suggestions! Now, passing the `onClose` method as a prop and removed the existing, now obsolete `onClose` callback in `JobStatusModal`.

**Review by benjdlambert — APPROVED** [review]

File: plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx [hunk 2]
```diff
@@ -46,17 +51,8 @@ export const JobStatusModal = ({ job, toCatalogLink }: Props) => {
     }
   };
 
-  const onClose = useCallback(() => {
-    if (!job) {
-      return;
-    }
-    if (job.status === 'COMPLETED' || job.status === 'FAILED') {
```
Comment (benjdlambert) [inline plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx:53]: This logic still needs to exist somewhere, as the onclose handler is called even when you accidentally click outside the modal. We don't want to close the modal accidentally.

**Review by benjdlambert — CHANGES_REQUESTED** [review]

File: plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx (hunk 2 above)
Comment (benjdlambert) [reply plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx:53]: This function will only close the modal if the job is completed or failed.

Comment (benjdlambert) [reply plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx:53]: Maybe we can move this state to the outside, this component which is just a prop saying `closable` or something?

Comment (OrkoHunter) [reply plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx:53]: Aha, so that's the catch! Knew something was up :P

Comment (OrkoHunter) [reply plugins/scaffolder/src/components/JobStatusModal/JobStatusModal.tsx:53]: Makes sense to keep this here as of now. Will revert bits of commit.

**Comment by OrkoHunter** [comment]
@benjdlambert Does this look good? The dialog is now not closable if the job is running.

**Comment by OrkoHunter** [comment]
Gentle ping @backstage/maintainers 🙇

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**